File: native_modules_demo/CustomToastModule.kt
package com.teacherapp.toast

import android.widget.Toast
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

// 15. Hafta Android Native Modül Örneği (Kotlin)
class CustomToastModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String {
        return "CustomToastModule"
    }

    @ReactMethod
    fun showToast(message: String) {
        Toast.makeText(reactApplicationContext, message, Toast.LENGTH_SHORT).show()
    }
}

File: native_modules_demo/CustomToastModule.swift
import Foundation
import UIKit

// 15. Hafta iOS Native Modül Örneği (Swift)
@objc(CustomToastModule)
class CustomToastModule: NSObject {

  @objc
  func showToast(_ message: String) {
    DispatchQueue.main.async {
      let alert = UIAlertController(title: "iOS Native Mesaj", message: message, preferredStyle: .alert)
      alert.addAction(UIAlertAction(title: "Tamam", style: .default, handler: nil))
      
      if let rootVC = UIApplication.shared.keyWindow?.rootViewController {
        rootVC.present(alert, animated: true, completion: nil)
      }
    }
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }
}